### PR TITLE
Tests for nbastats added

### PR DESCRIPTION
**Pull request by vanandsh**
- added tests for nbastats in tests/testthat/test-nbastats.R
- modified docstring in R/nbastats.R
- man files updated after running document()

### CODE REVIEW

File: R/nbastats.R [hunk 1]
```diff
@@ -3,24 +3,72 @@
...
+    if (length(playerNames) != 0) {
+      nba_data <- nba_data[nba_data$PLAYER %in% playerNames, ]
+    }
+
+    if (length(teamNames) != 0) {
+      nba_data <- nba_data[nba_data$Team %in% teamNames, ]
+    }
+
+    if (length(positions) != 0) {
+      nba_data <- nba_data[nba_data$POS %in% positions, ]
```
Comment (kfoofw) [inline R/nbastats.R]: Hi @vanandsh, as discussed verbally, the current `nbastats` function use may be complicated for the user. l

Let's modify the function to `nba_team_stats` where it no longer has a filter input based on player name. This new variant should be able to create descriptive statistics based on different teams. If any positions are specified as input, the filter should return descriptive stats for groupings of team AND positions.

Thanks for your hard work!

File: tests/testthat/test-nbastats.R [hunk 2]
```diff
@@ -1,3 +1,53 @@
...
+
+  test_that("Function does not throw an error when a proper dataframe is passed in data.", {
+    expect_error(
+      is.data.frame(nbastats('random values', columnNames = c('GP', '3PM', 'FT%'), all = FALSE))
+    )
+  })
+}
+
+data = read_csv("https://raw.githubusercontent.com/kfoofw/nba_espn/master/0.data/NBA_reg_2018-2019.csv")
+test_nbastats(data)
```
Comment (kfoofw) [inline tests/testthat/test-nbastats.R]: As discussed, please also modify the test scripts accordingly to the new variant of the function.

**Review by kfoofw — CHANGES_REQUESTED** [review]
preliminary review of `nbastats`

File: R/nbastats.R (hunk 1 above)
Comment (vanandsh) [reply R/nbastats.R]: Hi @kfoofw 

Thanks for the detailed review. I have made the changes requested for and tested it successfully on my local. Please review the latest changes.

Thanks
Anand

File: tests/testthat/test-nbastats.R (hunk 2 above)
Comment (vanandsh) [reply tests/testthat/test-nbastats.R]: @kfoofw  The test scripts have also been modified, and coverage is 100%.

File: tests/testthat/test-nba_team_stats.R [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+#'
+#' @param data tibble
+#' @export
+#'
+#' @importFrom readr read_csv
+#' @import dplyr
+#'
+#' @examples
+#' test_nbastats(nba_data)
+test_nbastats <- function(nba_data) {
```
Comment (kfoofw) [inline tests/testthat/test-nba_team_stats.R]:
```suggestion
test_nba_team_stats<- function(nba_data) {
```

File: tests/testthat/test-nba_team_stats.R [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  })
+  test_that("Function does not return correct number of descriptive stats columns when `stats_filter` and `positions_filter` are given", {
+    expect_equal(
+      ncol(nba_team_stats(nba_data, stats_filter = sample_stats_filter, positions_filter = sample_positions_filter)),
+      length(sample_stats_filter)*4 + 2
+    )
+  })
+}
+nba_data = read_csv("https://raw.githubusercontent.com/kfoofw/nba_espn/master/0.data/NBA_reg_2018-2019.csv")
+test_nbastats(nba_data)
```
Comment (kfoofw) [inline tests/testthat/test-nba_team_stats.R]:
```suggestion
test_nba_team_stats(nba_data)
```

**Review by kfoofw — CHANGES_REQUESTED** [review]
thanks @vanandsh for your edits! Some minor adjustments required. Can you also edit the main repo ReadMe.rmd (and md file) to change the function from `nbastats` to `nba_team_stats`, and the corresponding function description as well? Thanks!

**Review by kfoofw — APPROVED** [review]
thanks for the changes!

**Pull request merged**